SCClassLibrary/Common/Core/Clock.sc

```diff
@@ -395,3 +395,73 @@ elapsed time is whatever the system clock says it is right now. elapsed time is
 
 	archiveAsCompileString { ^true }
 }
+
+
+
+
+LinkClock : TempoClock {
+
+	var <>tempoChanged;
+
+	*newFromTempoClock { arg clock;
+		^super.new(
+			clock.tempo,
+			clock.beats,
+			clock.seconds,
+			clock.queue.maxSize
+		).initFromTempoClock(clock)
+	}
+
+	initFromTempoClock { arg clock;
+		var oldQueue;
+		//stop TempoClock and save its queue
+		clock.stop;
+		oldQueue = clock.queue.copy;
+		beatsPerBar = clock.beatsPerBar;
+
+		forBy(1, oldQueue.size-1, 3) {|i|
+			var task=oldQueue[i+1];
+			//change clock for Routines embeded inside PauseStream
+			if(task.isKindOf(PauseStream)){
```

Comment:
I am not sure why you think this doesn't work with other classes. It works well with `Function`, `Routine` or any other class I tested. The if statement for `PauseStream` is because it's [`awake` function](https://github.com/smiarx/supercollider/blob/3591f1d5ced6288011941afcc575dd4f59aa3629/SCClassLibrary/Common/Streams/Stream.sc#L417) calls its stream `beats_' function which uses the old clock, so we have to update it first.

Regarding objects with a `clock` property, I think the best way is to let them set it to the thread's clock when they are awoken, so you wouldn't have to bother setting it manually.
We can use this function to test this behaviour.
```supercollider
(
~teststream = { arg streamClass;
	var clock1 = TempoClock.new, clock2 = TempoClock.new;
	var stream = streamClass.new({
		loop {
			(stream.clock === clock2).postln;
			1.wait;
		}
	}).play(clock1);
	
	clock1.sched(2, {
		clock1.stop;
		clock2.sched(0, stream);
		nil
	});
};
)
```
If the `streamClass` respects this behaviour, the function should print true after the stream is scheduled by the second clock.

Routine already do this (see the primitive [RoutineResume](https://github.com/smiarx/supercollider/blob/3591f1d5ced6288011941afcc575dd4f59aa3629/lang/LangPrimSource/PyrPrimitive.cpp#L3290)), so `~teststream.value(Routine)` works.
`PauseStream` however doesn't. But if we change its `awake` function to
```supercollider
awake { arg beats, seconds, inClock;
	clock = inClock;
	^this.next(beats)
}
```
Now we have the same behaviour and `~teststream.value(Task)` works as expected.